### PR TITLE
Added code for wrapping INDArray with scala methods

### PR DESCRIPTION
**Pull request by hohonuuli**
I noticed that ND4J started work on a scala API. I wrote an example that uses very light-weight implicit conversions to wrap an existing INDArray. That way you don't need to create a scala specific version (i.e. ISNDArray). I wrote a few tests for you to demonstrate it's use too. I hope it's helpful for you.

Cheers
Brian

### CODE REVIEW

**Comment by hohonuuli** [comment]
whoops, just a note that in the pom.xml I forgot to set the scope of the _scalatest_ dependency to _test_.

File: nd4j-scala-api/src/test/scala/org/nd4j/api/linalg/INDArrayExtSpec.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package org.nd4j.api.linalg
+
+import org.junit.runner.RunWith
+import org.nd4j.linalg.factory.Nd4j
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{Matchers, FlatSpec}
+
+import Implicits._
```
Comment (hohonuuli) [inline nd4j-scala-api/src/test/scala/org/nd4j/api/linalg/INDArrayExtSpec.scala]: This line does all the magic. The conversion between the Scala DSL and nd4j API is essentially transparent to a user.

File: nd4j-scala-api/pom.xml [hunk 2]
```diff
@@ -48,6 +58,16 @@
...
+    <dependency>
+        <groupId>org.nd4j</groupId>
+        <artifactId>nd4j-jblas</artifactId>
+        <version>${project.version}</version>
+    </dependency>
+    <dependency>
+        <groupId>org.scalatest</groupId>
+        <artifactId>scalatest_${scala.version.short}</artifactId>
+        <version>${scalatest.version}</version>
+    </dependency>
```
Comment (hohonuuli) [inline nd4j-scala-api/pom.xml]: I forgot to set the scopes of these 2 new dependencies to _test_

File: nd4j-scala-api/src/test/scala/org/nd4j/api/linalg/INDArrayExtSpec.scala [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+@RunWith(classOf[JUnitRunner])
+class INDArrayExtSpec extends FlatSpec with Matchers {
+
+  "INDArrayEx" should "use the apply method to access values" in {
```
Comment (hohonuuli) [inline nd4j-scala-api/src/test/scala/org/nd4j/api/linalg/INDArrayExtSpec.scala]: Typo, that should be __INDArrayExt__ not _INDArrayEx__. It doesn't affect the test though, it's just for information purposes.

**Comment by agibsonccc** [comment]
Cool, I'll take a closer look at this soon. Just saw this.

**Comment by hohonuuli** [comment]
Great. I hope it's helpful. Let me know if you have any questions. Also, keep in mind it's just a start, there's more that could be done.

**Comment by agibsonccc** [comment]
Yeah definitely. I need to consolidate this with some of my changes as well.

**Comment by agibsonccc** [comment]
Merged commit. Thanks again!

**Pull request merged**